.github/workflows/tests.yml

```diff
@@ -25,6 +25,8 @@ jobs:
         python -m pip install --upgrade pip
         pip install .
         pip install pytest pytest-cov
+        pip install -r visualization/requirements.txt 
```

Comment:
I think it's best if you move this to a separate testing pipeline where you'd only run visualization tests. Your current tests fail most likely because you need to use some other plugin / base image to support selenium (I guess the current one does not have necessary requirements for browser-based tests). So it could be necessary to move these away anyway, but another reason to do that is that visualization is optional and we don't want to accidentally miss some import errors by installing these requirements for main tests.